### PR TITLE
Fix bug with MIME type warning on good labels

### LINKED ISSUE #429: validate warns "document standard id ... is not correct" on good labels
## 🐛 Describe the bug
Validate gives erroneous WARNINGs "document standard id ... is not correct"

## 📜 To Reproduce
Unzip the attachment, then
% validate -t EPPS_EDR_SIS.xml
[snip]
  PASS: file:/Users/rchen/Desktop/EPPS_EDR_SIS.xml
      WARNING  [warning.file.not_msword_mimetype]   file:/Users/rchen/Desktop/EPPS_EDR_SIS.DOC with document standard id 'Microsoft Word' is not correct
      WARNING  [warning.file.not_text_mimetype]   file:/Users/rchen/Desktop/EPPS_EDR_SIS.TXT with document standard id '7-Bit ASCII Text' is not correct
        1 product validation(s) completed

## 🕵️ Expected behavior
No validation errors

## 🩺 Test Data / Additional context
[Archive.zip](https://github.com/NASA-PDS/validate/files/7417071/Archive.zip)

## 🖥 System Info

---

## 🦄 Related requirements
:unicorn: #367

## ⚙️ Engineering Details
@qchaupds a few things looks like they need to be updated here:
- [ ] we are not being case agnostic in our file suffix patterns (see JPEG and MSWORD, as examples). let's just ignore case.

- [ ] text file check needs to be simplified and may just not be working
```
$ validate-2.2.0-SNAPSHOT/bin/validate -t src/test/resources/github367/document/invalid_pccref.xml
  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github367/document/invalid_pccref.xml
      WARNING  [warning.file.not_text_mimetype]   file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github367/document/pccref.txt with document standard id '7-Bit ASCII Text' is not correct
        1 product validation(s) completed

Summary:

  0 error(s)
  1 warning(s)

  Product Validation Summary:
    1          product(s) passed
    0          product(s) failed
    0          product(s) skipped

  Referential Integrity Check Summary:
    0          check(s) passed
    0          check(s) failed
    0          check(s) skipped

  Message Types:
    1            warning.file.not_text_mimetype

End of Report
Completed execution in 3650 ms
```
But why is that invalid? I even removed the file we had in the test resources and created my own and it says it is 
invalid [pccref.txt](https://github.com/NASA-PDS/validate/files/7419892/pccref.txt)?
```
$ file src/test/resources/github367/document/pccref.txt
src/test/resources/github367/document/pccref.txt: ASCII text
```
For text files, I think we should really simplify this check since there are really infinite number of file suffixes possible for text files (if they even have a file suffix). Let's just check that it is one of the more common image/document formats (check it does not match any of the other file suffixes, excluding HTML, Latex, Postcript, RTF since those are basically text so it doesn't really matter from an archive sense)

- [x] For the error messaging, can we change the wording to something like:
```
WARNING  [warning.file.not_msword_mimetype]   file:/Users/rchen/Desktop/EPPS_EDR_SIS.DOC with document standard 'Microsoft Word' is incorrect. Expected file suffix(es): [ doc, docx ]
```

- [x] For MSWord, we need to add additional file suffixes: docm, dot, dotm, dotx. For Excel we need to add: xltx, xlt, xlsm, xlsb

### PR DESCRIPTION
**Pull request by qchaupds**
1. Add test resources : src/test/resources/github429
2. Add two missing types 'DOC' and 'DOCX' for 'Microsoft Word' and 'XLS' and 'XLSX' for 'Microsoft Excel' to mime table src/main/resources/validate_default_mime_types.txt
3. Add new regression test : src/test/resources/features/validate.feature

Refs:

https://github.com/nasa-pds/validate/issues/429 validate warns "document standard id ... is not correct" on good labels

Closes https://github.com/nasa-pds/validate/issues/429 validate warns "document standard id ... is not correct" on good labels

Several missing MIME types for "Microsoft Word" were added to validate_default_mime_types.txt to fix the issue: DOC and DOCX
Several missing MIME types for "Microsoft Excel" were added to validate_default_mime_types.txt to fix the issue: XLS and XLSX

Tested in DEV:

Running validate against a valid label with three document types '.DOC', '.TXT' and '.XLS'.  No warning messages are expected.

```
% validate -R pds4.label -r report_github429_label_valid.json   -s json -t  src/test/resources/github429/EPPS_EDR_SIS.xml
 
```

### CODE REVIEW

**Comment by jordanpadams** [comment]
@qchaupds take a look at the description of #429 again. I include some details on design I believe we need here. I think we need to simplify the design a bit.

**Comment by qchaupds** [comment]
@jordanpadams Please have another look at the PR.

The latest checkin today should now addressed:

1. The check of file extension is now case insensitive.
2. Text file checking should work now.
3. The error message is now changed to include "Expected file suffix(es)".
4. New file extensions have been added to Microsoft Word and Microsoft Excel.

A new bad label has been added to allow to test a bad document product id: src/test/resources/github429/EPPS_EDR_SIS_bad.xml

Running it should have a warning and an improved error message:

validate -R pds4.label -r report_github429_label_invalid.json   -s json -t  src/test/resources/github429/EPPS_EDR_SIS_bad.xml

**Review by jordanpadams — CHANGES_REQUESTED** [review]
@qchaupds i'm not quite seeing the appropriate warnings. when i run against target EPPS_EDR_SIS.xml , I get these warning messages, and I had a couple questions:

* is this message saying EPPS_EDR_SIS.DOC is not really a Word Doc? the file suffix seems valid?
* `Expected file suffix(es): []` - is this supposed to indicate something?
* Can you take a look at the description for #429? I think there are a few sub-tasks I included there that have note been updated?

> Product Level Validation Results
> 
>   PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github429/EPPS_EDR_SIS.xml
>       WARNING  [warning.file.not_msword_mimetype]   file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github429/EPPS_EDR_SIS.DOC with document standard id 'Microsoft Word' is not correct.  Expected file suffix(es): []
>       WARNING  [warning.file.not_msexcel_mimetype]   file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github429/EPPS_EDR_SIS.XLS with document standard id 'Microsoft Excel' is not correct.  Expected file suffix(es): []
>         1 product validation(s) completed

**Comment by qchaupds** [comment]
@jordanpadams The label works for me in DEV.  You may have to do a "mvn clean" first because the MIME table (a resource) has been updated.  I think the issues have been addressed.  Please see the messages for the latest commit.  Which ones do you think have not?

```
{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 116 % validate -R pds4.label -r report_github429_label_valid.json   -s json -t  src/test/resources/github429/EPPS_EDR_SIS.xml  >& t6


{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 118 % grep file_name src/test/resources/github429/EPPS_EDR_SIS.xml
                <file_name>e.pdf</file_name>
                <file_name>EPPS_EDR_SIS.DOC</file_name>
                <file_name>EPPS_EDR_SIS.XLS</file_name>
                <file_name>EPPS_EDR_SIS.TXT</file_name>

The report did not show any warning .

{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 119 % egrep "status|message"  report_github429_label_valid.json
      "status": "PASS",
      "messages": [],
    "messageTypes": []

```

**Review by jordanpadams — CHANGES_REQUESTED** [review]
@qchaupds I provided some updates to the code and it now builds and executes correctly for the most part except PDF because the map you are using with the text file overwrites.

please checkout the updates I've made and test out on your own.

I would recommend re-evaluating using a text file, and instead use a JSON file with a dictionary that can be read in and maybe flipped around, e.g.

```
{ 
  standard1: [ extension1, extension2, etc ],
  standard2: [ extension1, extension2, etc ]
}
```

```
  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github429/EPPS_EDR_SIS.xml
      WARNING  [warning.file.not_html_mimetype]   file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github429/EPPS_EDR_SIS_1.foo with document standard 'PDF' is not correct.  Expected file suffix(es): []
        1 product validation(s) completed
```

**Comment by jordanpadams** [comment]
@qchaupds you can use Gson (already a dependency) to parse the JSON file where you are currently using a scanner and reading by tokens.

File: src/main/java/gov/nasa/pds/tools/util/DocumentUtil.java [hunk 1]
```diff
@@ -37,12 +41,109 @@
...
+  private ArrayList<String> docTypeList = new ArrayList<String>();
+  private ArrayList<ProblemType> problemTypeList = new ArrayList<ProblemType>();
+  private boolean classInitialized = false;  // Indicate whether if the mapping has been initialized or not. 
+
+  public DocumentUtil() {
+      // Because this class can be instantiated by other classes many times, we only want to call the initialize() function
+      // if we will be using the getProblemType() function below.
+  }
+
+  private boolean isClassInitialized() {
```
[the remark this answers is not in the document]
Comment (qchaupds) [reply src/main/java/gov/nasa/pds/tools/util/DocumentUtil.java:62]: @sonatype-lift ignore

**Pull request merged**